--- a/ldso/ldso/dl-elf.c
+++ b/ldso/ldso/dl-elf.c
@@ -317,7 +317,6 @@ struct elf_resolve *_dl_load_shared_library(unsigned int rflags, struct dyn_elf
 	   standard lib directories. */
 	_dl_if_debug_dprint("\tsearching multilib lib path list\n");
 	tpnt1 = search_for_named_library(libname, rflags,
-					UCLIBC_RUNTIME_PREFIX LDSO_MULTILIB_DIR ":"
 					UCLIBC_RUNTIME_PREFIX "usr/" LDSO_MULTILIB_DIR,
 					rpnt, NULL);
 	if (tpnt1 != NULL)
@@ -337,11 +336,7 @@ struct elf_resolve *_dl_load_shared_library(unsigned int rflags, struct dyn_elf
 	   This list must exactly match the list in uClibc/ldso/util/ldd.c */
 	_dl_if_debug_dprint("\tsearching full lib path list\n");
 	tpnt1 = search_for_named_library(libname, rflags,
-					UCLIBC_RUNTIME_PREFIX "lib:"
 					UCLIBC_RUNTIME_PREFIX "usr/lib"
-#ifndef __LDSO_CACHE_SUPPORT__
-					":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib"
-#endif
 					, rpnt, NULL);
 	if (tpnt1 != NULL)
 		return tpnt1;
--- a/utils/ldconfig.c
+++ b/utils/ldconfig.c
@@ -949,11 +949,8 @@ int main(int argc, char **argv)
 
 		/* look ma, no defaults */
 		if (!nodefault) {
-			scan_dir(UCLIBC_RUNTIME_PREFIX "lib");
 			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/lib");
-#ifndef __LDSO_CACHE_SUPPORT__
-			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib");
-#else
+#ifdef __LDSO_CACHE_SUPPORT__
 			/* I guess the defaults aren't good enough */
 			if ((extpath = get_extpath())) {
 				for (cp = strtok(extpath, DIR_SEP); cp; cp = strtok(NULL, DIR_SEP)) {
@@ -963,8 +960,7 @@ int main(int argc, char **argv)
 						while (cp[--len] == '/' && len)
 							cp[len] = 0;
 					/* we do the redundancy check only if cache usage is enabled */
-					if (strcmp(UCLIBC_RUNTIME_PREFIX "lib", cp) == 0
-					    || strcmp(UCLIBC_RUNTIME_PREFIX "usr/lib", cp) == 0) {
+					if (strcmp(UCLIBC_RUNTIME_PREFIX "usr/lib", cp) == 0) {
 						if (verbose >= 0)
 							warnx("You should remove `%s' from `%s'", cp, LDSO_CONF);
 						continue;
--- a/utils/ldd.c
+++ b/utils/ldd.c
@@ -504,10 +504,7 @@ static void locate_library_file(ElfW(Ehdr) *ehdr, ElfW(Dyn) *dynamic,
 
 	/* Lastly, search the standard list of paths for the library.
 	   This list must exactly match the list in uClibc/ldso/ldso/dl-elf.c */
-	path = UCLIBC_RUNTIME_PREFIX "lib:" UCLIBC_RUNTIME_PREFIX "usr/lib"
-#ifndef __LDSO_CACHE_SUPPORT__
-	    ":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib"
-#endif
+	path = UCLIBC_RUNTIME_PREFIX "usr/lib"
 	    ;
 	search_for_named_library(lib->name, buf, path);
 	if (*buf != '\0') {
